### PR TITLE
Make userfacing set_* functions generic

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
This PR makes the functions `set_engine()`, `set_mode()` and `set_args()` generic.

This is being done so [tidyclust](https://github.com/EmilHvitfeldt/tidyclust) can use these functions as well.

### CODE REVIEW

File: R/engines.R [hunk 1]
```diff
@@ -105,10 +105,12 @@ load_libs <- function(x, quiet, attach = FALSE) {
 #'
 #' @export
 set_engine <- function(object, engine, ...) {
+  UseMethod("set_engine")
+}
+
+#' @export
+set_engine.model_spec <- function(object, engine, ...) {
   mod_type <- class(object)[1]
-  if (!inherits(object, "model_spec")) {
```
Comment (simonpcouch) [inline R/engines.R:109]: Did we used to test these lines anywhere? Surprised there aren't any existing snaps or checks that needed adjusting.

**Review by simonpcouch — APPROVED** [review]
Looks great to me besides that one question about testing. :)

File: R/engines.R (hunk 1 above)
Comment (EmilHvitfeldt) [reply R/engines.R:109]: No prior tests: https://app.codecov.io/gh/tidymodels/parsnip/blob/master/R/engines.R
Screen Shot 2022-07-15 at 9 21 31 AM

Comment (simonpcouch) [reply R/engines.R:109]: Gotcha, sounds good👍

**Pull request merged**